### DISCUSSION TITLE
Is there a default open-actions.conf?

### DISCUSSION THREAD

**Discussion by elfehr** [kovidgoyal/kitty#5871] [Q&A] [answered]
Hello,

The reason for my question is that links to text files are opened with emacs instead of emacsclient, although $EDITOR is set to emacsclient (and for example `Ctrl+Shift+F2` correctly opens the config file in emacsclient). I tried writing my own `open-actions.conf` but I have trouble reproducing the default matching criteria. Can the default settings be found somewhere, so I could find what entrie(s) are invoking emacs and override only that?

**Comment by kovidgoyal** [comment, marked as answer]
You need to set editor system wide not in your shell, since the shell
runs after kitty. Or set editor in kitty.conf. If you want to see the
defaults look in open_action.py

On Mon, Jan 09, 2023 at 04:23:23PM -0800, elfehr wrote:
> Hello,
> 
> The reason for my question is that links to text files are opened with emacs instead of emacsclient, although $EDITOR is set to emacsclient (and for example `Ctrl+Shift+F2` correctly opens the config file in emacsclient). I tried writing my own `open-actions.conf` but I have trouble reproducing the default matching criteria. Can the default settings be found somewhere, so I could find what entrie(s) are invoking emacs and override only that?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5871
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by elfehr** [reply]
Thanks! I think I understood the problem: python's mimetypes module doesn't recognizes my text files as plain text files because they have no/unrecognized extensions.
It looks like kitty is defaulting to letting the OS open those file and what I took for the default matching criteria are actually the OS's default applications. My bad, I don't have a file explorer installed so I just never realized that the default OS behavior was to open emacs.

**Reply by simon-liebehenschel** [reply]
@elfehr Could you please share your `open-actions.conf`? Sadly, the documentation has no examples (except for macOS). I just want to open `<file_path>:<line_number>` file links in neovim.

**Reply by elfehr** [reply]
I ended up not needing any `open-actions.conf`, since the default were enough, so I have no example to share, sorry.
In case you didn't receive a notification when he answered, check out kovidgoyal's link in the thread below. I hope that can help!

**Comment by kovidgoyal** [comment]
Examples are at: https://sw.kovidgoyal.net/kitty/kittens/hyperlinked_grep/

**Comment by kovidgoyal** [comment]
On Thu, May 11, 2023 at 12:30:16PM +0000, AIGeneratedUsername wrote:
> @elfehr Thank you for your response. Actually, now I think that Kitty by itself can not help me with opening `<relative_file_path>:<line_number>` links from terminal output in my editor (neovim). I am switching to https://github.com/artemave/tmux_super_fingers which should support the described functionality.

Sure it can, https://sw.kovidgoyal.net/kitty/kittens/hints/
